src/vs/workbench/contrib/terminal/browser/terminalView.ts

```diff
@@ -110,6 +110,22 @@ export class TerminalViewPane extends ViewPane {
 				this._terminalTabbedView?.rerenderTabs();
 			}
 		}));
+		configurationService.onDidChangeConfiguration(e => {
+			if ((e.affectsConfiguration(TerminalSettingId.ShellIntegrationDecorationsEnabled) && !configurationService.getValue(TerminalSettingId.ShellIntegrationDecorationsEnabled)) ||
+				(e.affectsConfiguration(TerminalSettingId.ShellIntegrationEnabled) && !configurationService.getValue(TerminalSettingId.ShellIntegrationEnabled))) {
+				this._parentDomElement?.children[2].querySelectorAll(`.xterm`).forEach(e => e.classList.remove('shell-integration'));
```

Comment:
Can we set it on this element so we don't need to do it for every terminal? Simpler and this will probably fall apart when adding new terminals

<img width="421" alt="Screen Shot 2022-03-02 at 2 58 49 pm" src="https://user-images.githubusercontent.com/2193314/156463966-af3a26b2-1356-44c0-b5ba-d5206598ab2a.png">

Reply:
We need to adjust the left padding for each terminal though in order to make room for the decoration

Reply:
I mean doing this:

```css
.integrated-terminal.shell-integration-enabled .xterm {
	padding-left: 20px !important;
}
```

Instead of:

```css
.monaco-split-view2.vertical .split-view-view .xterm.shell-integration {
	padding-left: 20px !important;
}
```